src/index.ts

```diff
@@ -251,20 +280,31 @@ const PersistedQueriesPlugin: PostGraphilePlugin = {
   },
 
   // For regular HTTP requests
-  "postgraphile:httpParamsList"(paramsList, { options }) {
+  "postgraphile:httpParamsList"(paramsList, { options, req }) {
     return paramsList.map((params: any) => {
+      const allowUnpersistedOperations =
+        options.allowUnpersistedOperation || (() => false);
+
       // ALWAYS OVERWRITE, even if invalid; the error will be thrown elsewhere.
-      params.query = persistedOperationFromPayload(params, options) as string;
+      params.query = persistedOperationFromPayload(
+        params,
+        options,
+        allowUnpersistedOperations(req, params)
+      ) as string;
       return params;
     });
   },
 
   // For websocket requests
-  "postgraphile:ws:onOperation"(params, { message, options }) {
+  "postgraphile:ws:onOperation"(params, { message, options, req }) {
```

Comment:
We're not sent the `req` here; with hooks you have to search to see where they're called and what they're passed:

https://github.com/graphile/postgraphile/blob/ba9a7bcf1c3fa3347cf07de0a3732cfd0d5b6dcb/src/postgraphile/http/subscriptions.ts#L261-L266

However, we do have a helper for that:

https://github.com/graphile/postgraphile/blob/ba9a7bcf1c3fa3347cf07de0a3732cfd0d5b6dcb/src/postgraphile/http/subscriptions.ts#L405

which it turns out extracts req via:

```
    const req = socket['__postgraphileReq'];
```

So you should be able to get a req you can use:

```suggestion
  "postgraphile:ws:onOperation"(params, { message, options, socket }) {
    const req = socket['__postgraphileReq'];
```

Reply:
Thanks 👍🏼 I wouldn't have guessed without your help.